python/pyspark/sql/types.py

```diff
@@ -1034,11 +1089,19 @@ def _parse_datatype_json_value(json_value: Union[dict, str]) -> DataType:
             return _all_atomic_types[json_value]()
         elif json_value == "decimal":
             return DecimalType()
-        elif json_value == "timestamp_ntz":
-            return TimestampNTZType()
```

Comment:
This is unreachable. `if json_value in _all_atomic_types.keys():` above handles it.